src/app_service/service/mailservers/service.nim

```diff
@@ -49,18 +49,10 @@ const requestMoreMessagesTask: Task = proc(argEncoded: string) {.gcsafe, nimcall
   let arg = decode[RequestMoreMessagesTaskArg](argEncoded)
   try:
     info "Requesting additional message history for chat", chatId=arg.chatId
-    let response = status_mailservers.syncChatFromSyncedFrom(arg.chatId)
+    let response = status_mailservers.requestMoreMessages(arg.chatId)
 
     if(not response.error.isNil):
       error "Could not request additional messages due to error", errDescription = response.error.message
-
-    let syncedFrom = response.result.getInt()
-    if(syncedFrom != 0):
-      let resultJson = %* {
-        "chatId": arg.chatId,
-        "syncedFrom": syncedFrom
-      }
-      arg.finish(%resultJson)
```

Comment:
Without the `arg.finish`, the async task never finishes and the thread will not close.

If the RPC response it no longer returns anything, at least finish with `{"error": ""}`.

I see that the code is actually not very good and doesn't finish the task either if we have an exception. Maybe it's something you can improve, if not, I can open an issue to revisit them.

Reply:
I updated the code, but it's crashing now after `arg.finish` (if I remove arg.finish it doesn't crash), do you know what I am doing wrong?

Reply:
Most likely it is because slot is expecting a different data than you emit with `arg.finish`.
```
  proc mailserverSynced*(self: Service, syncInfo: string) {.slot.} =
    let syncInfoParsed = parseJson(syncInfo)
    let signalData = MailserverSyncedArgs(
      chatId: syncInfoParsed["chatId"].getStr(),
      syncedFrom: syncInfoParsed["syncedFrom"].getInt()
      )
    self.events.emit(SIGNAL_MAILSERVER_SYNCED, signalData)
```
I guess there is uncaught exception when handling `syncInfo`.

Reply:
thanks, that was the issue, i removed the signal since I don't think we used it, would that be ok?